### PR TITLE
Development

### PR DESCRIPTION
**Pull request by smohsinali**

### CODE REVIEW

File: test/test_util.py [hunk 1]
```diff
@@ -177,6 +178,28 @@ def test_random_neigborhood_conditional(self):
...
+
+        config = Configuration(diamond, {'bottom': 0, 'head': 0, 'left': 1, 'right': 1})
+        array = np.array([1., 1., 1., 0.])
+        hp_name = "head"
+        neighbor_value = 1
+
+        new_array = check_neighbouring_config(config, array, neighbor_value, hp_name)
+        expected_array = np.array([1, np.nan, np.nan, np.nan])
+
+        assert np.array_equal(new_array.astype(int), expected_array.astype(int))
```
Comment (mfeurer) [inline test/test_util.py]: Please do `np.testing.assert_almost_equal`.

File: ConfigSpace/util.py [hunk 2]
```diff
@@ -137,98 +233,99 @@ def get_one_exchange_neighbourhood(configuration: Configuration, seed: int) -> L
...
-                                        disabled.append(child_idx)
-                                        children = configuration.configuration_space._children_of[child]
-
-                                        for ch in children:
-                                            to_disable.add(ch.name)
-
-                    for idx in disabled:
-                        new_array[idx] = np.NaN
+                    new_array = check_neighbouring_config(configuration, new_array, neighbor_value, hp_name)
+                    # # Hyperparameters which are going to be set to inactive
```
Comment (mfeurer) [inline ConfigSpace/util.py]: No commented code please.

File: ConfigSpace/util.py [hunk 3]
```diff
@@ -71,6 +71,102 @@ def impute_inactive_values(configuration: Configuration, strategy: Union[str, fl
                                       allow_inactive_with_values=True)
     return new_configuration
 
+def check_neighbouring_config(configuration, new_array, neighbor_value, hp_name):
```
Comment (mfeurer) [inline ConfigSpace/util.py]: Could you please add type annotations?

File: test/test_util.py (hunk 1 above)
Comment (smohsinali) [inline test/test_util.py]: done

File: ConfigSpace/util.py (hunk 3 above)
Comment (smohsinali) [inline ConfigSpace/util.py]: done

File: ConfigSpace/util.py (hunk 2 above)
Comment (smohsinali) [inline ConfigSpace/util.py]: removed

**Pull request closed without merging**